### PR TITLE
Avoid deprecated notice when using symfony/config > 4.2

### PR DESCRIPTION
**Pull request by odolbeau**
See: https://github.com/symfony/symfony/blob/master/UPGRADE-4.2.md#config

### CODE REVIEW

**Comment by odolbeau** [comment]
@matthieumota: updated!

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -41,8 +42,9 @@ public function getConfigTreeBuilder()
      */
     private function getVendorNode($name)
     {
-        $treeBuilder = new TreeBuilder();
-        $node = $treeBuilder->root($name);
+        $treeBuilder = new TreeBuilder($name);
+        // Keep compatibility with symfony/config < 4.2
+        $rootNode = \method_exists(TreeBuilder::class, 'getRootNode') ? $treeBuilder->getRootNode() : $treeBuilder->root($name);
```
Comment (garethbraid89) [inline DependencyInjection/Configuration.php]:
```suggestion
        $node = \method_exists(TreeBuilder::class, 'getRootNode') ? $treeBuilder->getRootNode() : $treeBuilder->root($name);
```

File: DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -67,8 +69,9 @@ private function getVendorNode($name)
 
     private function getClassNode()
     {
-        $treeBuilder = new TreeBuilder();
-        $node = $treeBuilder->root('class');
+        $treeBuilder = new TreeBuilder('class');
+        // Keep compatibility with symfony/config < 4.2
+        $rootNode = \method_exists(TreeBuilder::class, 'getRootNode') ? $treeBuilder->getRootNode() : $treeBuilder->root('class');
```
Comment (garethbraid89) [inline DependencyInjection/Configuration.php]:
```suggestion
        $node = \method_exists(TreeBuilder::class, 'getRootNode') ? $treeBuilder->getRootNode() : $treeBuilder->root('class');
```

File: DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -121,8 +124,9 @@ private function getClassNode()
 
     private function getUploadableNode()
     {
-        $treeBuilder = new TreeBuilder();
-        $node = $treeBuilder->root('uploadable');
+        $treeBuilder = new TreeBuilder('uploadable');
+        // Keep compatibility with symfony/config < 4.2
+        $rootNode = \method_exists(TreeBuilder::class, 'getRootNode') ? $treeBuilder->getRootNode() : $treeBuilder->root('uploadable');
```
Comment (garethbraid89) [inline DependencyInjection/Configuration.php]:
```suggestion
        $node = \method_exists(TreeBuilder::class, 'getRootNode') ? $treeBuilder->getRootNode() : $treeBuilder->root('uploadable');
```

**Comment by odolbeau** [comment]
I didn't pay attention to that, thanks @garethbraid89!

**Comment by franmomu** [comment]
This bundle supports `"php": ">=5.3.2"` so `TreeBuilder::class` cannot be used because was introduced in `PHP 5.5.0`. So I would use https://github.com/doctrine/DoctrineBundle/pull/853 approach.

**Comment by matthieumota** [comment]
Okay so @odolbeau just need to pass treebuilder instance rather than FQN instruction.

**Comment by stof** [comment]
Actually closing because #388 also fixed it.

**Pull request closed without merging**